### PR TITLE
Auth: Add authentication config frontend routes to api.go

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
This PR adds the auth config routes to the `pkg/api/api.go` file.

**Why do we need this feature?**

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by Jguer — APPROVED** [review]

**Review by linoman — APPROVED** [review]

**Review by alexanderzobnin — APPROVED** [review]
Good catch!

File: pkg/api/api.go [hunk 1]
```diff
@@ -218,6 +218,12 @@ func (hs *HTTPServer) registerRoutes() {
 		r.Get("/user/auth-tokens/rotate", routing.Wrap(hs.RotateUserAuthTokenRedirect))
 	}
 
+	if hs.Features.IsEnabled(featuremgmt.FlagAuthenticationConfigUI) {
```
Comment (alexanderzobnin) [inline pkg/api/api.go:221]: Maybe we could move it into SAML package since it's enterpise only feature now?

**Review by alexanderzobnin — DISMISSED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (Jguer) [reply pkg/api/api.go:221]: `hs.Index` is specifically hard to register in enterprise, it's not a small fix unfortunately

Comment (alexanderzobnin) [reply pkg/api/api.go:221]: @Jguer @mgyongyosi we have this, for example in report package

Comment (Jguer) [reply pkg/api/api.go:221]: didn't see it already embedded the http server without dep cycles, that's nice :+1:

**Review by Jguer — APPROVED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (alexanderzobnin) [reply pkg/api/api.go:221]: Circular dependencies are funny 😄

**Review by linoman — APPROVED** [review]

File: public/app/routes/routes.tsx [hunk 2]
```diff
@@ -309,11 +309,14 @@ export function getAppRoutes(): RouteDescriptor[] {
...
       path: '/admin/authentication',
-      component: config.featureToggles.authenticationConfigUI
-        ? SafeDynamicImport(
-            () => import(/* webpackChunkName: "AdminAuthentication" */ 'app/features/auth-config/AuthConfigPage')
-          )
-        : () => <Redirect to="/admin" />,
+      component:
+        config.featureToggles.authenticationConfigUI &&
+        config.licenseInfo.enabledFeatures &&
+        config.licenseInfo.enabledFeatures['saml']
```
Comment (alexanderzobnin) [inline public/app/routes/routes.tsx:315]:
```suggestion
        config.licenseInfo.enabledFeatures?.saml
```

**Review by alexanderzobnin — APPROVED** [review]

**Pull request merged**